eox_tagging/edxapp_accessors.py

```diff
@@ -55,16 +58,33 @@ def get_course_enrollment(**kwargs):
     return CourseEnrollment.objects.get(user_id=user.id, course_id=course_id)
 
 
+def get_certificate(**kwargs):
+    """
+    Get GeneratedCertificate for specified id, key download URL or the course_id
+    and username associated with it.
+    """
+    username = kwargs.get("username")
+    course_id = kwargs.get("course_id")
+
+    return GeneratedCertificate.objects.get(
+        user__username=username,
+        course_id=CourseKey.from_string(course_id),
+    )
+
+
 def get_object(related_object_type, **kwargs):
     """Helper function to get objects using RELATED_FIELDS dictionary."""
-    RELATED_OBJECTS = {
+    related_objects_getter = {
         "user": get_user,
         "courseoverview": get_course,
         "site": get_site,
         "courseenrollment": get_course_enrollment,
+        "generatedcertificate": get_certificate,
     }
     try:
-        related_object = RELATED_OBJECTS.get(related_object_type.lower())(**kwargs)
+        related_object = related_objects_getter.get(
+            related_object_type.lower(),
+        )(**kwargs)
     except Exception:
```

Comment:
I removed catching the exception. If an exception is raised when creating a tag, then it is captured by the serializer. And if another service uses it, then it's better that raises the correct exception instead of ValidationError